src/routes/accounts.py

```
@@ -1,25 +1,198 @@
+
+
+@router.post("/activate/", status_code=200)
+async def activate(activation_data: UserActivationRequestSchema, db: AsyncSession = Depends(get_db)):
+    user = await get_user_by_email(db, activation_data.email)
+    token = await get_activate_token(db, activation_data.token)
+
+    if not token or token.expires_at < datetime.now():
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Invalid or expired activation token."
+        )
+    if user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="User account is already active."
+        )
+
+    await user_activation(db=db, user=user, token=token)
+
+    return {"message": "User account activated successfully."}
+
+
+@router.post("/password-reset/request/", status_code=200)
+async def password_reset_request(reset_data: PasswordResetRequestSchema, db: AsyncSession = Depends(get_db)):
+
+    user = await get_user_by_email(db, reset_data.email)
+
+    if not user or not user.is_active:
+        return {"message": "If you are registered, you will receive an email with instructions."}
+    else:
+        await deactivate_and_create_new_password_reset_token(db=db, user_id=user.id)
+
+    return {"message": "If you are registered, you will receive an email with instructions."}
+
+
+@router.post("/reset-password/complete/", status_code=200)
+async def password_reset_complete(reset_data: PasswordResetCompleteRequestSchema, db: AsyncSession = Depends(get_db)):
+    user = await get_user_by_email(db, reset_data.email)
+    if not user or not user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Invalid email or token."
+        )
+
+    db_token = await get_password_reset_token(db=db, user_id=user.id)
+
+    if reset_data.token != db_token.token :
+        await delete_token_with_exception(db, db_token)
```

Logic issue: If the provided token does not match, you delete the token but do not inform the user of the mismatch. According to the task requirements, you should raise an error or return a clear message when the token is invalid.